tools/istio-docker.mk

```diff
@@ -76,7 +85,7 @@ $(foreach FILE,$(DOCKER_FILES_FROM_ISTIO_BIN), \
         $(eval $(ISTIO_DOCKER)/$(FILE): $(ISTIO_BIN)/$(FILE) | $(ISTIO_DOCKER); cp $(ISTIO_BIN)/$(FILE) $(ISTIO_DOCKER)/$(FILE)))
 
 docker.sidecar_injector: BUILD_PRE=&& chmod 755 sidecar-injector
-docker.sidecar_injector: BUILD_ARGS=--build-arg BASE_VERSION=${BASE_VERSION}
+docker.sidecar_injector: BUILD_ARGS=--build-arg BASE_VERSION=${BASE_VERSION} ${DOCKER_OPT}
```

Comment:
all these changes don't feel right. I think there is a better way to do this by encoding `DOCKER_OPT` (which should probably be renamed to `BASE_IAMGE` in the makefile rule on line 285 of this file.